src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobMetadataStore.cs

```diff
@@ -85,25 +85,20 @@ public class BlobMetadataStore : IMetadataStore
 
         try
         {
-            await using (Stream stream = _recyclableMemoryStreamManager.GetStream(StoreInstanceMetadataStreamTagName))
-            await using (Utf8JsonWriter utf8Writer = new Utf8JsonWriter(stream))
+            await using Stream stream = _recyclableMemoryStreamManager.GetStream(StoreInstanceMetadataStreamTagName);
+            await JsonSerializer.SerializeAsync(stream, dicomDatasetWithoutBulkData, _jsonSerializerOptions, cancellationToken);
+
+            foreach (BlockBlobClient blob in blobClients)
             {
-                // TODO: Use SerializeAsync in .NET 6
-                JsonSerializer.Serialize(utf8Writer, dicomDatasetWithoutBulkData, _jsonSerializerOptions);
-                await utf8Writer.FlushAsync(cancellationToken);
-
-                foreach (var blob in blobClients)
-                {
-                    stream.Seek(0, SeekOrigin.Begin);
-                    await blob.UploadAsync(
-                        stream,
-                        new BlobHttpHeaders { ContentType = KnownContentTypes.ApplicationJson },
-                        metadata: null,
-                        conditions: null,
-                        accessTier: null,
-                        progressHandler: null,
-                        cancellationToken);
-                }
+                stream.Seek(0, SeekOrigin.Begin);
+                await blob.UploadAsync(
+                    stream,
+                    new BlobHttpHeaders { ContentType = KnownContentTypes.ApplicationJsonUtf8 },
```

Comment:
Ultimately, this is just more metadata for blobs:
![image](https://user-images.githubusercontent.com/2955179/176569791-c7520c3a-afdd-42f4-a294-fc2c93f326a3.png)